### PR TITLE
Hide output if Winget settings not existing

### PR DESCRIPTION
**Pull request by Romanitho**

### CODE REVIEW

**Comment by KnifMelti** [comment]
Trying this on a Windows Server 2019 Datacenter...
Microsoft.Powershell gets installed, but:
image
image

**Would be super nice if this was available!**

```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.



                888       888 d8b  .d8888b.           d8b
                888   o   888 Y8P d88P  Y88b          Y8P
                888  d8b  888     888    888
                888 d888b 888 888 888        888  888 888
                888d88888b888 888 888  88888 888  888 888
                88888P Y88888 888 888    888 888  888 888
                8888P   Y8888 888 Y88b  d88P Y88b 888 888
                888P     Y888 888  "Y8888P88  "Y88888 888

               https://github.com/Romanitho/Winget-Install
             https://github.com/Romanitho/Winget-Install-GUI

        _________________________________________________________


12:31:33 - ###   2023-10-30 - NEW INSTALL REQUEST   ###

12:31:33 - Running with admin rights.

12:31:33 - Checking prerequisites...
12:31:34 - Microsoft.VCLibs.140.00.UWPDesktop is not installed
12:31:34 - -> Downloading https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx...
12:31:34 - -> Installing Microsoft.VCLibs.140.00.UWPDesktop...
12:31:36 - -> Failed to intall Microsoft.VCLibs.140.00.UWPDesktop...
12:31:36 - WinGet is not installed
12:31:36 - -> Downloading Winget v1.6.2771
12:31:40 - -> Installing Winget v1.6.2771
12:31:51 - -> Winget installed.
12:31:51 - Checking prerequisites ended.

New-Item : Could not find a part of the path 'C:\Windows\System32\config\systemprofile\AppData\Local\Microsoft\WinGet\S
ettings\settings.json'.
At C:\ProgramData\winget-install.ps1:99 char:9
+         New-Item -Path $SettingsPath | Out-Null
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\Windows\Syst...s\settings.json:String) [New-Item], DirectoryNotFoundExce
   ption
    + FullyQualifiedErrorId : NewItemIOError,Microsoft.PowerShell.Commands.NewItemCommand

Out-File : Could not find a part of the path 'C:\Windows\System32\config\systemprofile\AppData\Local\Microsoft\WinGet\S
ettings\settings.json'.
At C:\ProgramData\winget-install.ps1:111 char:36
+ ... gFile | ConvertTo-Json | Out-File $SettingsPath -Encoding utf8 -Force
+                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OpenError: (:) [Out-File], DirectoryNotFoundException
    + FullyQualifiedErrorId : FileOpenFailure,Microsoft.PowerShell.Commands.OutFileCommand

12:31:51 - Start Microsoft.PowerShell processing...
12:31:53 - -> Microsoft.PowerShell exists on Winget Repository.
12:31:54 - -> Installing Microsoft.PowerShell...
12:31:54 - -> Running: "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.21.2771.0_x64__8wekyb3d8bbwe\winget.exe" install --id Microsoft.PowerShell -e --accept-package-agreements --accept-source-agreements -s winget -h


  ███▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 8.24 MB
  ███████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  2.00 MB / 8.24 MB
  ██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  3.00 MB / 8.24 MB
  ██████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  4.00 MB / 8.24 MB
  ██████████████████▒▒▒▒▒▒▒▒▒▒▒▒  5.00 MB / 8.24 MB
  █████████████████████▒▒▒▒▒▒▒▒▒  6.00 MB / 8.24 MB
  █████████████████████████▒▒▒▒▒  7.00 MB / 8.24 MB
  █████████████████████████████▒  8.00 MB / 8.24 MB
  ██████████████████████████████  8.24 MB / 8.24 MB

ReportIdentityFound PowerShell [Microsoft.PowerShell] ShowVersion 7.3.9.0
InstallationDisclaimer1
InstallationDisclaimer2
Downloading https://github.com/PowerShell/PowerShell/releases/download/v7.3.9/PowerShell-7.3.9-win-x64.msi



  █▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  4.00 MB /  100 MB
  ████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  14.0 MB /  100 MB
  ███████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  25.0 MB /  100 MB
  ██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  36.0 MB /  100 MB
  █████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  47.0 MB /  100 MB
  █████████████████▒▒▒▒▒▒▒▒▒▒▒▒▒  59.0 MB /  100 MB
  ███████████████████▒▒▒▒▒▒▒▒▒▒▒  67.0 MB /  100 MB
  ███████████████████████▒▒▒▒▒▒▒  78.8 MB /  100 MB
  ██████████████████████████▒▒▒▒  89.8 MB /  100 MB
  ██████████████████████████████   100 MB /  100 MB
  ██████████████████████████████   100 MB /  100 MB
InstallerHashVerified
InstallFlowStartingPackageInstall

InstallFlowInstallSuccess
12:32:28 - -> Microsoft.PowerShell successfully installed.
12:32:28 - Microsoft.PowerShell processing finished!

12:32:29 - ###   END REQUEST   ###

```

**Comment by KnifMelti** [comment]
Think yo'll have to `Out-Null` in the check also...

Current:
```
    #Check if setting file exist, if not create it
    if (Test-Path $SettingsPath) {
```

File: winget-install.ps1 [hunk 1]
```diff
@@ -96,7 +96,7 @@ function Add-ScopeMachine {
         $ConfigFile = Get-Content -Path $SettingsPath | Where-Object { $_ -notmatch '//' } | ConvertFrom-Json
     }
     else {
-        New-Item -Path $SettingsPath
+        New-Item -Path $SettingsPath | Out-Null
```
Comment (Romanitho) [inline winget-install.ps1:99]: I would even go for:
`New-Item -Path $SettingsPath -Force | Out-Null`

**Comment by Romanitho** [comment]
```
New-Item : Could not find a part of the path 'C:\Windows\System32\config\systemprofile\AppData\Local\Microsoft\WinGet\S
ettings\settings.json'.
At C:\ProgramData\winget-install.ps1:99 char:9
+         New-Item -Path $SettingsPath | Out-Null
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\Windows\Syst...s\settings.json:String) [New-Item], DirectoryNotFoundExce
   ption
    + FullyQualifiedErrorId : NewItemIOError,Microsoft.PowerShell.Commands.NewItemCommand
```

I would go for the `-force` anyway, just to be sure.

**Comment by KnifMelti** [comment]
Maybe disable Microsoft.VCLibs.140.00.UWPDesktop for server?

**Comment by Romanitho** [comment]
> image

Maybe you've run twice? And created by winget on first run.

**Comment by KnifMelti** [comment]
Have a winget app from SCCM now up and running on one of our ADM-servers thanks to your new winget scripts:
image

```
15:38:25 - Running with admin rights.
 
15:38:25 - Checking prerequisites...
15:38:26 - Microsoft.VCLibs.140.00.UWPDesktop is not installed
15:38:26 - -> Downloading https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx...
15:38:26 - -> Installing Microsoft.VCLibs.140.00.UWPDesktop...
15:38:27 - -> Failed to intall Microsoft.VCLibs.140.00.UWPDesktop...
15:38:27 - Installed Winget version: v1.6.2771
15:38:27 - Checking prerequisites ended.
15:38:27 - Start DominikReichl.KeePass processing...
15:38:28 - -> DominikReichl.KeePass exists on Winget Repository.
15:38:30 - -> Installing DominikReichl.KeePass...
15:38:30 - -> Running: "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.21.2771.0_x64__8wekyb3d8bbwe\winget.exe" install --id DominikReichl.KeePass -e --accept-package-agreements --accept-source-agreements -s winget -h --installer-type MSI --override "REBOOT=R /qn"
  ███▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 8.24 MB
  ███████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  2.00 MB / 8.24 MB
  ██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  3.00 MB / 8.24 MB
  ██████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  4.00 MB / 8.24 MB
  ██████████████████▒▒▒▒▒▒▒▒▒▒▒▒  5.00 MB / 8.24 MB
  █████████████████████▒▒▒▒▒▒▒▒▒  6.00 MB / 8.24 MB
  █████████████████████████▒▒▒▒▒  7.00 MB / 8.24 MB
  █████████████████████████████▒  8.00 MB / 8.24 MB
  ██████████████████████████████  8.24 MB / 8.24 MB
ReportIdentityFound KeePass [DominikReichl.KeePass] ShowVersion 2.55.0
InstallationDisclaimer1
InstallationDisclaimer2
Downloading https://sourceforge.net/projects/keepass/files/KeePass%202.x/2.55/KeePass-2.55.msi/download
  ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  0.00 B / 3.57 MB
  ████████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒  2.00 MB / 3.57 MB
  ████████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒  2.00 MB / 3.57 MB
  ██████████████████████████████  3.57 MB / 3.57 MB
  ██████████████████████████████  3.57 MB / 3.57 MB
InstallerHashVerified
InstallFlowStartingPackageInstall
InstallFlowInstallSuccess
15:39:20 - -> DominikReichl.KeePass successfully installed.
15:39:20 - DominikReichl.KeePass processing finished!
15:39:21 - ###   END REQUEST   ###
15:41:23 - ###   2023-10-30 - NEW UNINSTALL REQUEST   ###
 
15:41:23 - Running with admin rights.
 
15:41:23 - Checking prerequisites...
15:41:23 - Microsoft.VCLibs.140.00.UWPDesktop is not installed
15:41:23 - -> Downloading https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx...
15:41:24 - -> Installing Microsoft.VCLibs.140.00.UWPDesktop...
15:41:24 - -> Failed to intall Microsoft.VCLibs.140.00.UWPDesktop...
15:41:24 - Installed Winget version: v1.6.2771
15:41:24 - Checking prerequisites ended.
15:41:24 - Start DominikReichl.KeePass processing...
15:41:26 - -> Uninstalling DominikReichl.KeePass...
15:41:26 - -> Running: "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.21.2771.0_x64__8wekyb3d8bbwe\winget.exe" uninstall --id DominikReichl.KeePass -e --accept-source-agreements -h
  ███▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 8.24 MB
  ███████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  2.00 MB / 8.24 MB
  ██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  3.00 MB / 8.24 MB
  ██████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  4.00 MB / 8.24 MB
  ██████████████████▒▒▒▒▒▒▒▒▒▒▒▒  5.00 MB / 8.24 MB
  █████████████████████▒▒▒▒▒▒▒▒▒  6.00 MB / 8.24 MB
  █████████████████████████▒▒▒▒▒  7.00 MB / 8.24 MB
  █████████████████████████████▒  8.00 MB / 8.24 MB
  ██████████████████████████████  8.24 MB / 8.24 MB
ReportIdentityFound KeePass [DominikReichl.KeePass]
UninstallFlowStartingPackageUninstall
UninstallFlowUninstallSuccess
15:41:31 - -> DominikReichl.KeePass successfully uninstalled.
15:41:31 - DominikReichl.KeePass processing finished!
15:41:32 - ###   END REQUEST   ###

```

**Comment by Romanitho** [comment]
haha, you're welcome.

Don't know why `Microsoft.VCLibs.140.00.UWPDesktop` cannot be installed on servers as it is mandatory on workstations... I mean, maybe i'm confused with all my testings... is this really necessary even on workstations?
On recent tests, I was not able to install apps as system if `Microsoft.VCLibs.140.00.UWPDesktop` was not present.... 🤯

**Comment by Romanitho** [comment]
Maybe you're right, on server no need to try to install `Microsoft.VCLibs.140.00.UWPDesktop`

**Comment by KnifMelti** [comment]
...and after your cleanup etc. of [WAU ](https://github.com/Romanitho/Winget-AutoUpdate) we can install/use it on our 2019 management servers too (after deploying the **GPO** to them)!
The only error would be the installation of `Microsoft.VCLibs.140.00.UWPDesktop` - I'll test this as soon as I can...

**Comment by Romanitho** [comment]
Why do you need the GPO to block user context?
Install it full system only directly, no ?

**Comment by KnifMelti** [comment]
To avoid having multiple installations in SCCM

**Comment by Romanitho** [comment]
Not sure to understand

**Comment by KnifMelti** [comment]
We install WAU via an application in SCCM.
To avoid having one for standard clients and one for management servers only to block user context.

**Comment by Romanitho** [comment]
Ohhh, ok... got it.
One WAU, multiple conf by GPO 👌🏼

**Comment by KnifMelti** [comment]
One ring to rule them all 👌

**Pull request merged**